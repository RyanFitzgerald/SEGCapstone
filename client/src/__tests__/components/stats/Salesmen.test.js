import React from 'react';
import Salesmen from '../../../Components/stats/Salesmen';

describe('<Salesmen />', () => {

  beforeAll(() => {
    // Add necessary info to user
    const user = {
      role: {
        level: 1
      }
    };
    localStorage.setItem('user', JSON.stringify(user));

  });

  it('should render correctly', () => {
    // Return empty array on requests
    mock.onAny().reply(200, []);
    // Create mock functions / spies
    const setActiveSubtabMock = jest.fn();
    const renderErrorMock = jest.fn();
    const getTotalVolumeBySalesmenSpy = jest.spyOn(Salesmen.prototype, 'getTotalVolumeBySalesmen');

    // Shallow render it
    const wrapper = shallow(
      <Salesmen setActiveSubtab={setActiveSubtabMock} renderError={renderErrorMock}/>
    );

    // Compare to old snapshot
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    // Check set active subtab is called and with correct value
    expect(getTotalVolumeBySalesmenSpy).toHaveBeenCalledTimes(1);
    expect(setActiveSubtabMock).toHaveBeenCalledTimes(1);
    expect(setActiveSubtabMock.mock.calls[0][0]).toBe(3);
  });
});
