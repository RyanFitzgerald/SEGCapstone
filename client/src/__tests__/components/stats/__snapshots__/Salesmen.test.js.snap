// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Salesmen /> should render correctly 1`] = `<Loading />`;
